src/client/common/variables/environmentVariablesProvider.ts

```diff
@@ -58,12 +58,15 @@ export class EnvironmentVariablesProvider implements IEnvironmentVariablesProvid
     }
     public async getCustomEnvironmentVariables(resource?: Uri): Promise<EnvironmentVariables | undefined> {
         const workspaceFolderUri = this.getWorkspaceFolderUri(resource);
+        if (!workspaceFolderUri) {
+            return;
+        }
         this.trackedWorkspaceFolders.add(workspaceFolderUri ? workspaceFolderUri.fsPath : '');
 
         // eslint-disable-next-line
         // TODO: This should be added to the python API (or this entire service should move there)
         // https://github.com/microsoft/vscode-jupyter/issues/51
-        const envFile = workspaceFolderUri?.fsPath ? path.join(workspaceFolderUri.fsPath, '.env') : '.env';
```

Comment:
This was bogus, we cannot have a file named `.env` that is not a valid file (unless it exists in the users root directory, doesn't work on windows thought).
I.e its an invalid case, here we're expecting fully qualified file paths.